bot/cogs/reddit.py

```diff
@@ -47,6 +53,61 @@ def channel(self) -> TextChannel:
         """Get the #reddit channel object from the bot's cache."""
         return self.bot.get_channel(Channels.reddit)
 
+    async def get_access_tokens(self) -> None:
```

Comment:
Would it be better to utilise the 1-hour expiry of the tokens to not refresh them *every* time a command is run, instead only if a command is run and the access token is older than an hour? For me, if I were to check the top command, I would probably check the weekly and daily at the same time - even if I was just testing/seeing what the command does. This may reduce some unnecessary token regeneration, keeping the command a little faster.

Something like
```py
# setting:
self.access_token = (token, datetime.utcnow() + timedelta(hours=1))

# retrieving
if self.access_token[1] > datetime.utcnow():
    # regenerate
return self.access_token[0]
```
Could use a named tuple if you're feeling fancy.

Reply:
Agree, I'll change it.